journal.py

```diff
@@ -92,45 +103,94 @@ def home(request):
     return {'entries': entries}
 
 
-@view_config(route_name='add', renderer='templates/add.jinja2')
-def add(request):
+def do_add(request, redirect):
     if request.method == 'POST':
         title = request.params.get('title')
         text = request.params.get('text')
         if not (title == "" or text == ""):
             Entry.write(title=title, text=text)
```

Comment:
If you look carefully at the implementation of this method, what does it return?  What does that mean for the rest of your function body below?